src/framework/FormatChecker.ts

```diff
@@ -50,46 +47,43 @@ export class FormatChecker extends Base {
     });
 
     if (this.checker instanceof RegExp) {
-      if (this.checker.test(message.cleanContent) === true) return;
-    } else if (this.checker instanceof Function) {
-      if (this.checker(message.cleanContent, logger)) return;
-    }
+      if (this.checker.test(message.cleanContent)) return;
+    } else if (this.checker(message, logger)) return;
 
     logger.debug('bad format detected');
 
     const { cleanContent, author } = message;
 
     await message.delete();
     const warningContent = [
-      'Bonjour,',
       `Le message que vous avez posté dans ${message.channel} est incorrectement formaté, il a donc été supprimé.`,
       'Pour rappel, voici le message que vous aviez envoyé :',
-      `\`\`\`${cleanContent}\`\`\``,
+      `\`\`\`\n${cleanContent}\`\`\``,
+      // à simplifier :
       ...(this.examples !== undefined
         ? [
             `Voici ${this.examples.length > 1 ? 'des' : 'un'} example${
               this.examples.length > 1 ? 's' : ''
-            } de message correctement formaté :`,
-            ...this.examples.map((example) => `\`\`\`${example}\`\`\``),
+            } de message${
+              this.examples.length > 1 ? 's' : ''
+            } correctement formaté${this.examples.length > 1 ? 's' : ''} :`,
+            ...this.examples.map((example) => `\`\`\`\n${example}\`\`\``),
```

Comment:
Stock la présence ou l'absence dans une variable, pas besoin de refaire `this.examples.length > 1 ? 's' : ''` à chaque fois.

Reply:
Je sais (c'est en rapport avec la review d'avant). Je n'ai pas d'idée pour le nom (de la constante), j'ai préféré du coup laisser ça.
Des idées ?

Reply:
`plural` ira très bien je pense, sauf si quelqu'un a une meilleure idée.

Reply:
J'y ai pensé, mais ça fait un peu vague. Je vais faire avec.